### ISSUE TITLE
Serialize the data array

### ISSUE THREAD

**Issue by ppetzold** [ppetzold/nestjs-paginate#80]
Original: https://github.com/ppetzold/nestjs-paginate/issues/72#issuecomment-908314018

> With serialization, I mean filtering and transforming the response data. Using [class-transformer](https://github.com/typestack/class-transformer), for example.
> 
> For now, in my code, I have implemented an interceptor (also could be a class or helper function) to catch the Paginated and call the `plainToClass()` for data param. Here I can provide a small example for the better understanding:
> 
> ```
> // controller
> 
> const paginatedResults = await paginate(...) // Item[]
> return {
>     ...paginatedResults,
>     data: plainToClass(ItemResponseDto, paginatedResults.data)
> }
> ```
> 
> ```
> // ItemResponseDto with class-transformer
> 
> @Exclude()
> export class ItemResponseDto {
>   @Expose()
>   id: string;
> ...
> }
> ```
> 
> Having written this, I think this logic should be the responsibility of this module. So, nevermind 😅

**Comment by ricbermo** [comment]
I was able to workaround that using the following code 
![image](https://user-images.githubusercontent.com/2601363/145621732-94399463-c622-4167-804e-7f17678f5ebe.png)

**Comment by ppetzold** [comment]
Yup. That would be an example for the query builder solution I was referring to in #114 

This issue here tho refers more to the incorporation of all feature of `class-transformer` (https://github.com/typestack/class-transformer#additional-data-transformation). Besides hiding/exposing properties, it has decorators for additional data transformation etc.

**Comment by Joesilgos** [comment]
**Thanks for the example — I was banging my head on this!**  
I came here to check if there was something already open or resolved, and ended up fixing it like this.


Recently, I ran into an issue when trying to use `class-transformer` decorators like `@Exclude()` and `@Expose()` on my entities.

Since `paginate()` returns entity instances directly, the decorators aren't applied automatically, so things like hidden fields or exposed getters don’t show up as expected.

I solved it with a simple workaround in my service:

```ts
// after calling paginate()
paginated.data = instanceToPlain(paginated.data)
```
What this does:
- Uses the entity instances already returned by paginate
- Applies @Exclude() and @Expose() 

This workaround is working perfectly until there's native support for transformation

Suggestion: it would be great to have a built-in option like serializer: true, or to internally apply instanceToPlain() to automatically respect class-transformer decorators.

I’ll try to find some time to contribute — this library is just too good!
